src/core/viewports/orbit-viewport.js

```diff
@@ -43,30 +45,35 @@ export default class OrbitViewport extends Viewport {
     translationY = 0, // in pixels
     zoom = 1
   }) {
-    const eye = vec3_add([], lookAt, [0, 0, distance]);
-    vec3_rotateX(eye, eye, lookAt, rotationX / 180 * Math.PI);
-    vec3_rotateY(eye, eye, lookAt, rotationY / 180 * Math.PI);
+    const rotationMatrix = mat4_rotateX([], createMat4(), -pitchAngle / 180 * Math.PI);
+    if (orbitAxis === 'Z') {
+      mat4_rotateZ(rotationMatrix, rotationMatrix, -orbitAngle / 180 * Math.PI);
+    } else {
+      mat4_rotateY(rotationMatrix, rotationMatrix, -orbitAngle / 180 * Math.PI);
+    }
+
+    const transformMatrix = mat4_translate([], createMat4(),
+      [translationX / width * 2, translationY / height * 2, 0]);
+    mat4_scale(transformMatrix, transformMatrix, [zoom, zoom, zoom]);
+    mat4_multiply(rotationMatrix, rotationMatrix, transformMatrix);
```

Comment:
Nit: store final result `transformMatrix` as it can contain translations and rotations.